### PR TITLE
MD-98 Cria formulário para criar empresa

### PR DESCRIPTION
**Pull request by ebertonArouxa**

### CODE REVIEW

File: src/app/mocks/imageMock.ts [hunk 1]
```diff
@@ -0,0 +1 @@
...BBAAgQAghAAAIQAAEMABBBAAgQAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAAYQAABBAAAEEMAQAgQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABCAAAEEEMAQAgQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEEEMAQAgQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABBBAAgQAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABCAAAAEQQQQAABBBAgAghAAAIQAAEMABBBAA`;
```
Comment (claudiocassimiro) [inline src/app/mocks/imageMock.ts:1]: If you not use this const,  you can remove it

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+vi.mock(`next/navigation`, () => {
+  const actual = vi.importActual(`next/navigation`);
+  return {
+    ...actual,
+    useRouter: vi.fn(() => ({
+      push: vi.fn(),
+    })),
+  };
+});
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:18]: Without this mock the tests runs? If yes, you can remove it

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+  afterAll(() => {
+    vi.clearAllMocks();
+  });
+
+  afterEach(() => {
+    cleanup();
+    vi.resetAllMocks();
+  });
+
+  it(`When trying to send form without filling the company name input should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:49]:
```suggestion
  it(`When trying to send form without filling the company name input should, render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    });
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O Nome do restaurante é obrigatório.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form without filling the CNPJ/CPF input should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:63]:
```suggestion
  it(`When trying to send form without filling the CNPJ/CPF input, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    render(<CreateCompany />);
+
+    const submitButton = screen.getByRole(`button`, {
+      name: `Finalizar cadastro`,
+    });
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O Nome do restaurante é obrigatório.`)
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:59]:
```suggestion
      await screen.findByText(`O Nome do restaurante é obrigatório.`)
```
you have to remove the "*" char of all findByTexts of the test file

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+    await userEvent.type(companyNameInput, `restaurante`);
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O CNPJ/CPF é obrigatório e deve ser válido.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form filling a wrong CNPJ/CPF should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:80]:
```suggestion
  it(`When trying to send form filling a wrong CNPJ/CPF, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(cnpjInput, `123456789123`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O CNPJ/CPF é obrigatório e deve ser válido.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form without filling the phone number input should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:102]:
```suggestion
  it(`When trying to send form without filling the phone number input, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(cnpjInput, `12345678912`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O Telefone é obrigatório.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form filling a wrong phone number should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:124]:
```suggestion
  it(`When trying to send form filling a wrong phone number, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(phoneNumberInput, `219999999999`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O número de telefone deve ser válido.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form filling a wrong delivery phone number should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:142]:
```suggestion
  it(`When trying to send form filling a wrong delivery phone number, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(
+        `*O número de telefone deve ser válido ou estar vázio.`
+      )
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form without logo image should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:163]:
```suggestion
  it(`When trying to send form without logo image, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(phoneNumberInput, `21999999999`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*A logo é obrigatória.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form without filling the zipCode input should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:189]:
```suggestion
  it(`When trying to send form without filling the zipCode input, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(phoneNumberInput, `21999999999`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O CEP é obrigatório.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form filling a wrong zipCode should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:215]:
```suggestion
  it(`When trying to send form filling a wrong zipCode, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(zipCodeInput, `999999999`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*Digite um CEP válido.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form without filling the address input should render error`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:233]:
```suggestion
  it(`When trying to send form without filling the address input, should render error`, async () => {
```

File: src/app/createCompany/tests/createCompany.test.tsx [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+    await userEvent.type(zipCodeInput, `99999999`);
+
+    await userEvent.click(submitButton);
+
+    expect(
+      await screen.findByText(`*O Endereço é obrigatório.`)
+    ).toBeInTheDocument();
+  });
+
+  it(`When trying to send form filling all inputs correctly should change page`, async () => {
```
Comment (claudiocassimiro) [inline src/app/createCompany/tests/createCompany.test.tsx:263]:
```suggestion
  it(`When trying to send form filling all inputs correctly, createCompany func should have been called 1 time`, async () => {
```

File: src/app/createCompany/page.tsx [hunk 15]
```diff
@@ -1,18 +1,216 @@
...
+  const handleFileChange = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+    setFileName: Dispatch<SetStateAction<string>>,
+    setFile: Dispatch<SetStateAction<string>>
+  ) => {
+    const file = event.target.files?.[0];
+    setFileName(String(file?.name));
+    const fileToBase64 = await convertFileToBase64(file);
+    setFile(String(fileToBase64));
+    setLogoImageError(``);
```
Comment (claudiocassimiro) [inline src/app/createCompany/page.tsx:60]:
```suggestion
    try {
       const fileToBase64 = await convertFileToBase64(file);
       setFile(String(fileToBase64));
       setLogoImageError(``);
    } catch (e) {
       setLogoImageError(`Erro ao converter a imagem para pase 64`);
    }
```

File: src/app/createCompany/page.tsx [hunk 16]
```diff
@@ -1,18 +1,216 @@
...
+              Icon={<FaPhoneVolume size={20} color="#9ca3af" />}
+              name="deliveryPhoneNumber"
+              labelText="Telefone de delivery"
+              error={errors.deliveryPhoneNumber?.message}
+              register={register(`deliveryPhoneNumber`)}
+              placeholder="Somente números, com DDD"
+              type="number"
+            />
+            <p className={styles.companyInformationText}>
+              **Caso não seja passado, usaremos o telefone
```
Comment (claudiocassimiro) [inline src/app/createCompany/page.tsx:148]:
```suggestion
              Caso não seja passado, usaremos o telefone
```

File: src/app/createCompany/page.tsx [hunk 17]
```diff
@@ -1,18 +1,216 @@
...
+              placeholder="Ex: Rua 1, nº 1, casa 1 - Bairro luz"
+            />
+          </div>
+          <ButtonSubmit
+            isSubmiting={isSubmiting}
+            text="Finalizar cadastro"
+            submitError={requestError}
+            className={styles.companySubmitButton}
+            onClick={() => {
+              setLogoImageError(selectedLogo ? `` : `*A logo é obrigatória.`);
```
Comment (claudiocassimiro) [inline src/app/createCompany/page.tsx:209]:
```suggestion
              setLogoImageError(selectedLogo ? `` : `A logo é obrigatória.`);
```

File: src/app/components/UploadImageInput/styles.module.css [hunk 18]
```diff
@@ -15,5 +19,9 @@
 }
 
 .fileUploadedName {
-  @apply w-full h-full text-xs text-gray-500 py-1 px-1;
+  @apply w-full h-full text-xs text-white py-1 px-1;
```
Comment (claudiocassimiro) [inline src/app/components/UploadImageInput/styles.module.css:22]:
```suggestion
  @apply w-full h-full text-xs text-white p-1;
```

**Review by EricksonSiqueira — APPROVED** [review]

**Pull request merged**